### PR TITLE
istioctl x precheck

### PR DESCRIPTION
**Pull request by esnible**
Resolves https://github.com/istio/istio/issues/20355

In 1.5.x and earlier, "istioctl verify-install" did two things.  If it had an argument it verified that Istio was installed, if it didn't it verified if Istio *could be installed*.

This doesn't work for multiple control planes and was confusing.

With this PR:
We now have "istioctl x precheck", which does the existing pre-Istio-install cluster compatibility check.
The existing check now supports multiple control planes -- you can ask if your cluster is compatible with an IOP, even if Istio is already installed.
The existing "istioctl verify-install" does not do pre-install check.

This is just the basic support.  We don't check compatibility with the IOP flags except for "revision" and "namespace".  We don't support `--set` to modify the IOP being checked.  If there is interest I will do those in a follow-up.

cc @irisdingbj

### CODE REVIEW

File: istioctl/pkg/install/pre-check.go [hunk 1]
```diff
@@ -300,3 +319,186 @@ func (c *preCheckClient) checkMutatingWebhook() error {
...
+			if err != nil {
+				return err
+			}
+			if !nsExists {
+				return installPreCheck(targetNamespace, kubeConfigFlags, c.OutOrStdout())
+			}
+
+			// The Istio namespace does exist, but it wasn't installed by 1.6.0+ because no
+			// IstioOperator is there.
+			c.Printf("Istio already installed in namespace %q\n.  Skipping pre-check.", targetNamespace)
```
Comment (irisdingbj) [inline istioctl/pkg/install/pre-check.go:413]: Move `\n` in the end:
c.Printf("Istio already installed in namespace %q.  Skipping pre-check . \n ", targetNamespace)
Now it is a little strange:
```
Istio already installed in namespace "istio-system"
.  Skipping pre-check.Use 'istioctl upgrade' to upgrade or 'istioctl install --set revision=<revision>' to install another control plane.
```

File: istioctl/pkg/install/pre-check.go [hunk 2]
```diff
@@ -300,3 +319,186 @@ func (c *preCheckClient) checkMutatingWebhook() error {
...
+		retval = append(retval, istioInstall{
+			namespace: iop.Namespace,
+			revision:  iop.Spec.Revision,
+		})
+	}
+	return retval, nil
+}
+
+// nolint: lll
+func getIOPFromFile(restClientGetter resource.RESTClientGetter, options resource.FilenameOptions) (*operator_v1alpha1.IstioOperator, error) {
```
Comment (irisdingbj) [inline istioctl/pkg/install/pre-check.go:446]: If the cluster is not aware of the IOP crd, this method will fail like:
`unable to recognize "/Users/isisding/oti/goProjects/istio.io/istio/operator/data/profiles/iris.yaml": no matches for kind "IstioOperator" in version "install.istio.io/v1alpha1"`

Comment (esnible) [reply istioctl/pkg/install/pre-check.go]: fixed

File: istioctl/pkg/install/pre-check.go (hunk 1 above)
Comment (esnible) [reply istioctl/pkg/install/pre-check.go]: fixed

**Pull request merged**